### PR TITLE
Add 'Contributing' section to README

### PR DESCRIPTION
**Pull request by Gudahtt**
This section includes the test and release information that was present already, but adds a "Setup" section to explain how to setup the development environment. It also explains the `yarn setup` command, which is a common point of confusion for new contributors.

Linting instructions have been added alongside the test instructions as well.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,13 +18,23 @@ _Add examples here_
...
 
-Run `yarn test` to run the tests once.
+### Setup
 
-To run tests on file changes, run `yarn test:watch`.
+- Install [Node.js](https://nodejs.org) version 12
+  - If you are using [nvm](https://github.com/creationix/nvm#installation) (recommended) running `nvm use` will automatically choose the right node version for you.
+- Install [Yarn v1](https://yarnpkg.com/en/docs/install)
+- Run `yarn setup` to install dependencies and run any requried post-install scripts
+  - **Warning**: Do not use the `yarn` / `yarn install` command directly. Use `yarn setup` instead. The normal install command will skip required post-install scripts, leaving your development environment in an invalid state.
```
Comment (rekmarks) [inline README.md:29]:
```suggestion
  - **Warning:** Do not use the `yarn` / `yarn install` command directly. Use `yarn setup` instead. The normal install command will skip required post-install scripts, leaving your development environment in an invalid state.
```

**Review by rekmarks — APPROVED** [review]
LGTM! I just made on tiny copy tweak.

File: README.md (hunk 1 above)
Comment (Gudahtt) [reply README.md]: Hmm, interesting, I never know whether to bold the colon. This seems fine though.

Comment (rekmarks) [reply README.md]: I think it just looks (and is conceptually) more cohesive if the entire thing is bolded, as opposed to just the word. Pick your own adventure:

**Warning**: Foo.

vs

**Warning:** Foo.

Comment (rekmarks) [reply README.md]: A different angle: why _shouldn't_ the colon also be bolded?

**Pull request merged**